sdk/python/feast/pyspark/abc.py

```diff
@@ -66,6 +70,27 @@ def cancel(self):
         """
         raise NotImplementedError
 
+    def to_proto(self) -> JobProto:
+        """Converts SparkJob to the protobuf object.
+
+        Returns:
+            JobProto: Converted protobuf object.
+        """
+        job = JobProto()
+        job.id = self.get_id()
+        status = self.get_status()
+        if status == SparkJobStatus.COMPLETED:
```

Comment:
What the difference between SparkJobStatus & JobStatusProto?
Why can't we have only proto version used everywhere instead of SparkJobStatus and omit this mapping?

Reply:
Let's keep `SparkJob` a pure abstract class, even if have to duplicate a couple lines of code. This version is substantially less readable to me than the previous one, since now `JobProto` construction logic is spread across the class hierarchy.

Reply:
@pyalex i agree seems like there no reason why we shouldn't unify those. I'd address this in a separate PR though (after all, that enum wasn't introduced or even changed in this PR).